extensions/amp-list/0.1/test/test-amp-list-load-more.js

```diff
@@ -0,0 +1,125 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpDocService} from '../../../../src/service/ampdoc-impl';
+import {AmpList} from '../amp-list';
+import {Services} from '../../../../src/services';
+import {isExperimentOn, toggleExperiment} from '../../../../src/experiments';
+
+
+describes.realWin('amp-list component', {
+  amp: {
+    ampdoc: 'single',
+    extensions: ['amp-list'],
+  },
+}, env => {
+  let win, doc, ampdoc, sandbox;
+  let element, list;
+  let templates;
+
+  beforeEach(() => {
+    win = env.win;
+    doc = win.document;
+    ampdoc = env.ampdoc;
+    sandbox = env.sandbox;
+
+    templates = {
+      findAndSetHtmlForTemplate: sandbox.stub(),
+      findAndRenderTemplate: sandbox.stub(),
+      findAndRenderTemplateArray: sandbox.stub(),
+    };
+    sandbox.stub(Services, 'templatesFor').returns(templates);
+    sandbox.stub(AmpDocService.prototype, 'getAmpDoc').returns(ampdoc);
+
+    toggleExperiment(win, 'amp-list-load-more', true);
+  });
+
+  describe('manual', () => {
+
+    beforeEach(() => {
+      expect(isExperimentOn(win, 'amp-list-load-more')).to.be.true;
+
+      element = doc.createElement('amp-list');
+      list = new AmpList(element);
+
+      sandbox.stub(list, 'getAmpDoc').returns(ampdoc);
+      sandbox.stub(list, 'getFallback').returns(null);
+
+      sandbox.stub(list, 'mutateElement').callsFake(mutator => {
+        mutator();
+        return Promise.resolve();
+      });
+
+      sandbox.stub(list, 'measureElement').callsFake(measurer => {
+        measurer();
+        return Promise.resolve();
+      });
+
+      sandbox.stub(list, 'measureMutateElement').callsFake(
+          (measurer, mutator) => {
+            measurer();
+            mutator();
+            return Promise.resolve();
+          });
+
+      element.setAttribute('src', '/list/infinite-scroll?items=2&left=1');
+      element.setAttribute('load-more', 'manual');
+      element.setAttribute('layout', 'fixed');
+      element.setAttribute('width', '50');
+      element.setAttribute('height', '10');
+
+      element.style.height = '10px';
+      doc.body.appendChild(element);
+
+      sandbox.stub(list, 'getOverflowElement').returns(null);
+      sandbox.stub(list, 'fetchList_').returns(Promise.resolve());
+      list.element.changeSize = () => {};
+      list.buildCallback();
+    });
+
+    it('should create load-more elements after layout callback', async() => {
+      sandbox.stub(list, 'getPlaceholder').returns(null);
+      await list.layoutCallback();
+      expect(list.element.querySelector('[load-more-button]')).to.be.ok;
```

Comment:
Might be nice to be a little more general and do:

```js
expect(list.element.querySelectorAll('[load-more-button]')).to.have.length(1);
```

So that you can make sure two of them weren't added in some weird flow.